### PR TITLE
Security Logging

### PR DESCRIPTION
**Pull request by lyie28**
back up secu + Ophelie changes to tutorial

### CODE REVIEW

File: server/bastionlab_polars/src/composite_plan.rs [hunk 1]
```diff
@@ -116,6 +121,35 @@ impl CompositePlan {
             ));
         }
 
+        let mut inputs = String::from("");
```
Comment (dhalf) [inline server/bastionlab_polars/src/composite_plan.rs:124]: Put this in a dedicated function or method

File: server/bastionlab_polars/src/lib.rs [hunk 2]
```diff
@@ -86,6 +90,53 @@ impl BastionLabPolars {
...
+    fn update_log(
+        self: &BastionLabPolars,
+        new_row: DataFrame,
+        ) -> Result<(), Status> {
+        let mut dfs = self.dataframes.write().unwrap(); 
+        let artifact = dfs.get_mut(LOG_DF);
+        match artifact{
+            Some(artifact) => { //log dataframe exists, append new_row and save
+                artifact.dataframe = artifact.dataframe.vstack(&new_row).unwrap_or(artifact.dataframe.clone());
+                drop(dfs);
```
Comment (dhalf) [inline server/bastionlab_polars/src/lib.rs:102]: Usually you don't need to explicitly drop something in Rust.
Why is this needed here? If this is just a matter of scope, use a block {}

File: server/bastionlab_polars/src/lib.rs [hunk 3]
```diff
@@ -86,6 +90,53 @@ impl BastionLabPolars {
...
+                };
+                let df = DataFrameArtifact::new(new_row, pol, Vec::new());
+                dfs.insert(LOG_DF.to_string(), df);
+                drop(dfs);
+                return self.persist_df(LOG_DF);
+            }
+        }    
+    }
+
+    fn update_log_value(
```
Comment (dhalf) [inline server/bastionlab_polars/src/lib.rs:118]: What is this method doing?
Ideally, you just need to append new lines to the log
I can't see a situation that requires to update a past log, it seems to go against the very point of having logs

File: server/bastionlab_polars/src/lib.rs [hunk 4]
```diff
@@ -431,6 +487,24 @@ impl PolarsService for BastionLabPolars {
             identifier.clone()
         );
 
+        //LAURA LOG
```
Comment (dhalf) [inline server/bastionlab_polars/src/lib.rs:490]: Same, wrap in a dedicated function or method

File: server/bastionlab_polars/src/lib.rs [hunk 5]
```diff
@@ -443,11 +517,48 @@ impl PolarsService for BastionLabPolars {
         let fut = {
             let df = self.get_df(
                 &request.get_ref().identifier,
-                Some(self.sess_manager.get_client_info(token)?),
+                Some(self.sess_manager.get_client_info(token.clone())?),
             )?;
             stream_data(df, 32)
         };
-        Ok(fut.await)
+        let ret = fut.await;
```
Comment (dhalf) [inline server/bastionlab_polars/src/lib.rs:524]: Consider wrapping as well

**Review by dhalf — CHANGES_REQUESTED** [review]

**Review by dhalf — CHANGES_REQUESTED** [review]
Remove .collect when using the LOG_DF, it is already fetchable!

File: server/bastionlab_polars/src/lib.rs [hunk 6]
```diff
@@ -443,11 +566,24 @@ impl PolarsService for BastionLabPolars {
...
+
+        //logging
+        let user_hash= self.sess_manager.get_user_id(token.clone())?;
+        let id = &request.get_ref().identifier[..];
+        let pol = self.log_get_pol(id);
```
Comment (dhalf) [inline server/bastionlab_polars/src/lib.rs:578]: Use better variable names

**Review by dhalf — CHANGES_REQUESTED** [review]

File: server/bastionlab_polars/src/composite_plan.rs [hunk 7]
```diff
@@ -150,6 +150,14 @@ impl CompositePlan {
     }
 }
 
+impl Deref for CompositePlan {
```
[the remark this answers is not in the document]
Comment (lyie28) [reply server/bastionlab_polars/src/composite_plan.rs:153]: This was so I could get an iterator of the CompositePlan for my function which iterates over the composite plan and gets the entrypoints for the inputs column in log because I couldn't get an iter or access to composite_plan.0 from lib.r. I see that I can put that function into CompositePlan instead and call it from lib.rs (and so delete Deref).
